### PR TITLE
Sync feature spec

### PR DESCRIPTION
**Pull request by skiflink**
This PR involves increasing the coverage of the test suite.. by starting to add some feature specs.

Originally I was against this because I wanted to keep the test suite as quick as possible, but feature specs are *ESSENTIAL*. Without feature specs, you just have to hope that you've covered every scenario in your service specs and everything _just works_. 

Once I created my feature spec, added Sepia (HTTP interaction recorder), and ran.. it was clear that without feature specs, I'd be running a faulty application and I wouldn't have even known!

Running a feature spec that tested the syncing feature as a whole, found that:
* My shop count was breaking completely (one instance where the count was not being passed back properly)
* My database functions weren't escaping text as they should be (it seems I've gotten too used to using ORMs).

This feature spec will also save my back in the future as I update the sync and refactor it so it's nice and clean. Code coverage is essential, and you will learn this when you need ~100% uptime applications. You can lose a lot of uptime, just by a simple bug that you didn't pick up in your test, basically because you were lacking in code coverage or you aren't doing enough expectations.

TLDR; Added a feature spec for my syncing feature.. found some errors I didn't find through service specs. I have a lot more confidence in my application now, and this is only the beginning.

### CODE REVIEW

**Comment by skiflink** [comment]
UPDATE:

There are a few TODO's before this is ready for merge

- [ ] A quick revise on my db inserts: Escaping characters for SQL
- [ ] Add a service spec scenario to test against the error I found from feature spec
- [ ] _If I get time_ Add maps key as environment variable, otherwise do this very soon!

**Comment by skiflink** [comment]
Going to implement https://github.com/hiddentao/squel to combat issues with current sql query builder, and without going full on sequelize

**Comment by skiflink** [comment]
Going to implement https://www.npmjs.com/package/dotenv for google maps api key.. Will add an example .env
